url71665d7d68_branscha_lib_scripty_tgz-pJ8-com_sdicons_scripty_ScriptyAutoReplJ8/src/com/sdicons/scripty/parser/Parser.java

```diff
@@ -0,0 +1,386 @@
+                        return new Token(Token.TokenType.eof, String.format("Unclosed string starting on line: %d col: %d with contents \"%s\".", lLine, lCol, lValue.toString()), lLine, lCol);
+                    }
+                    else 
+                    {
+                        // The string ended in a normal way.
+                        // We skip the " delimiter without doing anything with it.
+                        aBuffer.consumeChar();
+                        return new Token(Token.TokenType.string, lValue.toString(), lLine, lCol);
+                    }
+                }
+                default:
+                {
+                    if(Character.isWhitespace(lChar))
+                    {                
+                        // Whitespace encountered. In this mode we gobble all whitespace and put it in 
+                        // a single token.
+                        final StringBuilder lValue = new StringBuilder();
+                        lValue.append(lChar);
+                        while (!aBuffer.eof() && Character.isWhitespace(aBuffer.peekChar()))
+                            lValue.append(aBuffer.consumeChar());
+                        // We return the accumulated whitespace.
+                        return new Token(Token.TokenType.whitespace, lValue.toString(), lLine, lCol);
+                    } 
+                    else 
+                    {
+                        // Normal identifier encountered. We turn this into a string as well.
+                        // We have to test on set of characters that do not appear in any of the
+                        // previous branches.
+                        final StringBuilder lValue = new StringBuilder();
+                        lValue.append(lChar);
+                        char lPeek = aBuffer.peekChar();
+                        while (!aBuffer.eof() && '(' != lPeek && ')' != lPeek && '"' != lPeek && !Character.isWhitespace(lPeek) && '=' != lPeek && ';' != lPeek)
+                        {
+                            lValue.append(aBuffer.consumeChar());
+                            lPeek = aBuffer.peekChar();
+                        }
+                        return new Token(Token.TokenType.string, lValue.toString(), lLine, lCol);
+                    }
+                }
+            }
+        }        
+    }
+    
+    private  void pushBackToken(Token lToken)
+    {
+        pushback = lToken;
+    }
+    
+    private Token getNextNonWhitespaceToken(IParserInput aBuffer)
+    {
+        Token lToken = getNextToken(aBuffer);
+        while(lToken.isWhitespace()) lToken = getNextToken(aBuffer);        
+        return lToken;
+    }
+    
+    /**
+     * Returns a token or a List<Object> ...
+     * A token result always indicates an error, parsing failed.
+     * 
+     * @param aBuffer
+     * @return
+     */
+    @SuppressWarnings("unchecked")
+	private Object parseList(IParserInput aBuffer)
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 12. By comparison, 98% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.